js/views/SurveyListPage.js

```diff
@@ -35,11 +37,22 @@ const SurveyListPage = React.createClass({
   },
 
   componentDidMount() {
-    this._surveys = loadCachedSurveyList().slice();
-    if (this._surveys.length === 0) {
-      loadSurveyList(this.loadList);
+    NetInfo.fetch().done((reach) => {
```

Comment:
I like this idea.  However, it does hide the fact that async network operations are pending (regardless if they will fail because the user if offline).  If the user doesn't have network then maybe we shouldn't even bother making the request in the first place?  Ex: L51 `loadSurveyList({forceRefresh: true}` is a network request.